packages/configurationmap/components/HashMap.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+
+import ConfigurationMapSection from "./ConfigurationMapSection";
+import ConfigurationMapHeading from "./ConfigurationMapHeading";
+import ConfigurationMapRow from "./ConfigurationMapRow";
+import ConfigurationMapLabel from "./ConfigurationMapLabel";
+import ConfigurationMapValue from "./ConfigurationMapValue";
+
+const METHODS_TO_BIND = ["formatValue", "isHashMap"];
+
+type headingLevel = 1 | 2 | 3 | 4 | 5 | 6;
+
+interface HashMapProps {
+  /**
+   * Content to render if no value is available
+   */
+  emptyValue?: string;
+  /**
+   * The data to render. Keys are rendered as labels, values are rendered as the value
+   */
+  hash: { [key: string]: React.ReactNode };
+  /**
+   * Priority of the heading that the `headline` prop renders in. Numbers map to <h1> through <h6>
+   */
+  headingLevel?: headingLevel;
```

Comment:
is there a concrete need for `headingLevel` already?

my gut says that the library should decide on this most or all of the time.

Reply:
I've seen a bunch of cases in `dcos-ui` where we manually set the `headingLevel`, so I figured I'd keep that here.